### ISSUE TITLE
Feature request: Alpha clone character presets

### ISSUE THREAD

**Issue by Cribbit** [pyfa-org/Pyfa#716]
Would be useful for designing fittings for alpha clone characters.

**Comment by blitzmann** [comment]
What is an alpha clone character? Can you please describe?

**Comment by Cribbit** [comment]
You haven't heard yet? Eve is going to have alpha clones, free to play players with a very limited set of skills trainable and most only to III or IV. https://community.eveonline.com/news/dev-blogs/introducing-clone-states-and-the-future-of-access-to-eve-online/

**Comment by blitzmann** [comment]
What the fuck

**Comment by blitzmann** [comment]
No, I haven't heard. I haven't actually played the game in over a year. =3

I'll have to look through this and figure out what it actually does. I'll keep this issue up to date

**Comment by Cribbit** [comment]
https://docs.google.com/spreadsheets/d/1zSAmxqlqyWjPNxgfadJD8gSIwDFw58i6bPD7NA7AwIU/edit should be helpful.

**Comment by blitzmann** [comment]
thanks!

**Comment by Ebag333** [comment]
@blitzmann IIRC CCP is planning on exposing these skills via CREST.  So it should be relatively easy for us to write something to nab the skills and create characters off them.

We can hand create the characters when the patch hits, but CCP is planning on fiddling with the skills, so I wouldn't count on it being static by any measure.

**Comment by blitzmann** [comment]
Yeah... we could even have a flag on the character itself, so that we don't have new "alpha" characters along side you existing character. Which makes sense, since you're still using your character for the fit, just with the restricted skill set.

Skills are normally applied via the `skill.level` attribute, which is already a getter. We can simply add logic in there to determine if we have to restrict the skillset. We would would also need to define the character as a certain race.

I like the possibilities here, and it really shouldn't be too difficult. The only thing is managing that list of skills and updating when we noticed they change.

**Comment by Ebag333** [comment]
>Yeah... we could even have a flag on the character itself, so that we don't have new "alpha" characters along side you existing character. Which makes sense, since you're still using your character for the fit, just with the restricted skill set. So if you want to see the alpha version, you just set the character as an alpha.

CCP did something very similar initially and people _hated_ it on SiSi.  I think it opens up a whole realm of potential issues where someone's reporting that things aren't working correctly, and it turns out they had the alpha checkbox checked.

Plus if alpha skills change, that could get really complex in a hurry.

I think a cleaner method is to have a separate char just like we do for 0 and V.  We would be adding at least 4 chars, possibly 8 (if we also can add starting skill chars).  Okay, so that's 10 chars, which means you're going to freak out about how much room that takes up. :)

We can tweak the drop down so that those go to the bottom of the list, so users chars float to the top and it's not in their way.

By keeping them separate, we can more cleanly manage them because their skills _will_ change over time, so we can do whatever we want (like completely nuking and rebuilding them), where as if we're modifying a users character that becomes a lot trickier.

**Comment by blitzmann** [comment]
> CCP did something very similar initially and people hated it on SiSi. I think it opens up a whole realm of potential issues where someone's reporting that things aren't working correctly, and it turns out they had the alpha checkbox checked.

I mean, that's just how it is. We can do things like change the skillbook icon from red to yellow if they are on an alpha clone (or appending `(ALPHA)` to the character name, which might be better).

> I think a cleaner method is to have a separate char just like we do for 0 and V. We would be adding at least 4 chars, possibly 8 (if we also can add starting skill chars). Okay, so that's 10 chars, which means you're going to freak out about how much room that takes up. :) I

Yep, freaking out. I don't agree at all. Not only does it clutter up the default characters list, but consider this: You're a gallente pilot, but trained into minmatar. Your sub lapsed. So no more minmatar for you, only gallente. We don't want a caharacter with the max alpha levels, we want the players character with the shit gallente levels, for proper representation on what an alpha character does to your fit.

> where as if we're modifying a users character that becomes a lot trickier.

We would basically be setting caps on the skill levels based on the alpha flag, so no character modification. And sure, the alpha skills will change over time, but that's just something we'll have to watch and react to. I don't particularly like the idea of the client going out and fetching that information from EVE independently, although I guess that is an option.

**Comment by Ebag333** [comment]
>We don't want a caharacter with the max alpha levels, we want the players character with the shit gallente levels, for proper representation on what an alpha character does to your fit.

This is a fair point.  I still don't like actually modifying the character's levels, and we will still have people asking for the default characters (so they can design fits for them).  We already have people asking for starting char skills, and there's been multiple mentions about alpha char skills, so I'm not sure I see a way around it.

Um.  We might be able to do something to flag skills as being Alpha/Omega.

>I don't particularly like the idea of the client going out and fetching that information from EVE independently, although I guess that is an option.

This seems to come up a lot.  I don't like the idea of Pyfa going the way of EFT just because someone isn't around to do a build.  (And yes, I know it's open source, but I swear if you get hit by a bus tomorrow we're all effed.)

**Comment by blitzmann** [comment]
> We already have people asking for starting char skills, and there's been multiple mentions about alpha char skills, so I'm not sure I see a way around it.

That's a separate issues, please don't lump them into this one. :)

>  I don't particularly like the idea of Pyfa going the way of EFT just because someone isn't around to do a build.

And this seems to come up a lot. It's not something to worry about. Why don't you give an example on how we could handle this then, without making multiple default characters because that isn't going to happen (because clutter and because that's not how people are going to expect it to work).

If you're concerned about maintainability, default characters is not the way to go. It isn't dynamic at all, and if a skill level changes for an alpha character, we have to create migrations and not fuck up the users database in the process.

> We might be able to do something to flag skills as being Alpha/Omega. 

I'm hoping this is the way CCP implements it - as an attribute on the skill itself. That would make life much easier.

**Comment by blitzmann** [comment]
>  I still don't like actually modifying the character's levels, 

Also, as stated... we're not modifying the levels. We're setting an effective cap. So... v0v

**Comment by MaxGriswald** [comment]
So I posted on another suggestion before posting here.  I have to say I DO NOT agree with blitzmann about the generic characters.  Yes, it does add a little clutter to the dropdown if you add generic Alpha characters, but if I am trying to help a Caldari corpmate fit a ship and I am a Minmatar, just flagging my character as Alpha for fitting isn't going to do a damn thing to help.  Same as if I am starting a new character and want to know what the best race is to work on if I am going to be Alpha.  For example, setting up a Gnosis.  What race can make the most of the Gnosis?  Unless I had four max'd Alpha characters already, I can't know that if it is just a toggle for an actual account.

I don't see any reason to add starter characters as a default, since anyone can just create a character to find the skills for it.  But for determining the maximum potential of an Alpha character, the only ways to do that is either spend a couple months maxing the character out before realizing there was a better race to start as, or having a non-API character on the dropdown.  If you have a non-API character on the dropdown, it might as well be put on there one time when you download the pyfa program thereby saving thousands of people the effort of adding it.  Just my two cents (And it isn't because I am too lazy to do it, because I already added them to my own install, it's just that it would be nice to save other people time as well, because it took me almost an hour to get it set up for all four races.)
